File: 배열/프로그래머스lv1-제일 작은 수 제거하기.py
def solution(arr):
    if not arr or len(arr) == 1:
        return [-1]
    
    min_number = min(arr)
    return [x for x in arr if x != min_number]
"""
정확성  테스트
테스트 1 〉	통과 (1.89ms, 15.8MB)
테스트 2 〉	통과 (0.02ms, 10.2MB)
테스트 3 〉	통과 (0.03ms, 10.3MB)
테스트 4 〉	통과 (0.02ms, 10.2MB)
테스트 5 〉	통과 (0.01ms, 10.2MB)
테스트 6 〉	통과 (0.03ms, 10.4MB)
테스트 7 〉	통과 (0.05ms, 10.2MB)
테스트 8 〉	통과 (0.00ms, 10.2MB)
테스트 9 〉	통과 (0.01ms, 10.1MB)
테스트 10 〉	통과 (0.00ms, 10.3MB)
테스트 11 〉	통과 (0.01ms, 10.2MB)
테스트 12 〉	통과 (0.01ms, 10.3MB)
테스트 13 〉	통과 (0.01ms, 10.1MB)
테스트 14 〉	통과 (0.03ms, 10.2MB)
테스트 15 〉	통과 (0.02ms, 10.1MB)
테스트 16 〉	통과 (0.04ms, 10.2MB)

"""

File: 배열/프로그래머스lv1-제일 작은 수 제거하기.js
function solution(arr) {
  if (arr.length === 0 || arr.length === 1) return [-1]

  const minNumber = Math.min(...arr)
  return arr.filter((x) => x !== minNumber)
}
// 정확성  테스트
// 테스트 1 〉	통과 (2.67ms, 37.6MB)
// 테스트 2 〉	통과 (0.07ms, 30MB)
// 테스트 3 〉	통과 (0.07ms, 30.3MB)
// 테스트 4 〉	통과 (0.12ms, 29.9MB)
// 테스트 5 〉	통과 (0.07ms, 30.3MB)
// 테스트 6 〉	통과 (0.08ms, 30.3MB)
// 테스트 7 〉	통과 (0.08ms, 30.3MB)
// 테스트 8 〉	통과 (0.06ms, 29.9MB)
// 테스트 9 〉	통과 (0.05ms, 29.9MB)
// 테스트 10 〉	통과 (0.05ms, 30MB)
// 테스트 11 〉	통과 (0.09ms, 30MB)
// 테스트 12 〉	통과 (0.06ms, 29.9MB)
// 테스트 13 〉	통과 (0.08ms, 30.1MB)
// 테스트 14 〉	통과 (0.14ms, 30.3MB)
// 테스트 15 〉	통과 (0.09ms, 30.3MB)
// 테스트 16 〉	통과 (0.11ms, 30.3MB)
